Makefile

```diff
@@ -34,7 +39,8 @@ clean_docs: ## Clean the documentation build
 
 .PHONY: build_pypi
 build_pypi: ## Build the Python package
-	python setup.py sdist bdist_wheel
+	pip install build
```

Comment:
Shouldn't the version of `build` be specified in the `pyproject.toml` too?

Reply:
Actually think it makes sense to do the build (and publish) via a github action, in order to give is complete control over the build environment. Have added a `.github/workflows/publish.yml` (still requires testing/refinement). 

This approach also simplifies the release process to something like:

1. `git checkout master` in current local repo (no need for fresh virtual env etc. `setuptools_scm` will check for dirty workdir). 
2. Update `README.md` and citation. 
3. Final `git commit` and `git tag`.
4. Push the commit and tag. Github will build and push.

As a safety net we can have logic to push to TestPyPI if tagged with RC version, otherwise push to PyPI.